### PR TITLE
update mcoreddp call

### PR DESCRIPTION
**Pull request by akoumpa**
# What does this PR do ?
Updates Mcore DDP call to match https://github.com/NVIDIA/Megatron-LM/commit/02871b4df8c69fac687ab6676c4246e936ce92d0

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by akoumpa** [comment]
Draft because it's pending mcore update on GH

**Review by timmoon10 — COMMENTED** [review]
Overall looks good, but we should update the Megatron-LM commit in [Dockerfile](https://github.com/NVIDIA/NeMo/blob/55a97386fff1f0d01fc998e0071cebb851ed313a/Dockerfile#L70), [Dockerfile.ci](https://github.com/NVIDIA/NeMo/blob/55a97386fff1f0d01fc998e0071cebb851ed313a/Dockerfile.ci#L37), and [README](https://github.com/NVIDIA/NeMo/blob/55a97386fff1f0d01fc998e0071cebb851ed313a/README.rst?plain=1#L363).

**Comment by akoumpa** [comment]
Thank you @timmoon10 for the fast review, will update the PR.

File: Dockerfile [hunk 1]
```diff
@@ -67,7 +67,7 @@ WORKDIR /workspace/
 RUN git clone https://github.com/NVIDIA/Megatron-LM.git && \
   cd Megatron-LM && \
   git checkout 36e9b6bf3d8034b10c9bbd9fc357c2df2bd1515c && \
-  git cherry-pick -n e69187bc3679ea5841030a165d587bb48b56ee77 && \
+  git cherry-pick -n 02871b4df8c69fac687ab6676c4246e936ce92d0 && \
```
Comment (timmoon10) [inline Dockerfile:70]: We should either cherry-pick both https://github.com/NVIDIA/Megatron-LM/commit/e69187bc3679ea5841030a165d587bb48b56ee77 and https://github.com/NVIDIA/Megatron-LM/commit/02871b4df8c69fac687ab6676c4246e936ce92d0, or we should just checkout https://github.com/NVIDIA/Megatron-LM/commit/02871b4df8c69fac687ab6676c4246e936ce92d0 directly:
```suggestion
  git checkout 02871b4df8c69fac687ab6676c4246e936ce92d0 && \
```

Comment (akoumpa) [reply Dockerfile]: You are right, sorry about that, fixed it.

**Review by timmoon10 — APPROVED** [review]

**Pull request merged**